Revert the following commit:

commit b660e95a13cd6f53aad2ec34a34de7df6a18a2a3
Author: Takuto Ikuta <tikuta@chromium.org>
Date:   Mon Mar 30 01:42:11 2026 -0700

    build: Use tool inputs instead of siso config for rust actions
    
    Bug: 491242305
    Change-Id: Id476f75efeb47eca7435b09befb1be47ab05f95f
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/7707348
    Auto-Submit: Takuto Ikuta <tikuta@chromium.org>
    Reviewed-by: Junji Watanabe <jwata@google.com>
    Commit-Queue: Junji Watanabe <jwata@google.com>
    Commit-Queue: Takuto Ikuta <tikuta@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#1606959}

--- a/build/config/siso/rust.star
+++ b/build/config/siso/rust.star
@@ -197,6 +197,11 @@ def __step_config(ctx, step_config):
         # TODO(b/285225184): use precomputed subtree
         "third_party/rust-toolchain:toolchain",
     ]
+    rust_inputs = [
+        "build/action_helpers.py",
+        "build/gn_helpers.py",
+        "build/rust/gni_impl/rustc_wrapper.py",
+    ] + rust_toolchain
     rust_link_indirect_inputs = {
         "includes": [
             # https://crbug.com/488158799#comment21 explains why `rustc` requires
@@ -240,7 +245,7 @@ def __step_config(ctx, step_config):
         {
             "name": "rust_bin",
             "action": "(.*_)?rust_bin",
-            "inputs": rust_toolchain + clang_inputs,
+            "inputs": rust_inputs + clang_inputs,
             "indirect_inputs": rust_link_indirect_inputs,
             "handler": "rust_link_handler",
             "deps": "none",  # disable gcc scandeps
@@ -251,7 +256,7 @@ def __step_config(ctx, step_config):
         {
             "name": "rust_cdylib",
             "action": "(.*_)?rust_cdylib",
-            "inputs": rust_toolchain + clang_inputs,
+            "inputs": rust_inputs + clang_inputs,
             "indirect_inputs": rust_link_indirect_inputs,
             "handler": "rust_link_handler",
             "deps": "none",  # disable gcc scandeps
@@ -262,7 +267,7 @@ def __step_config(ctx, step_config):
         {
             "name": "rust_macro",
             "action": "(.*_)?rust_macro",
-            "inputs": rust_toolchain + clang_inputs,
+            "inputs": rust_inputs + clang_inputs,
             "indirect_inputs": rust_link_indirect_inputs,
             "handler": "rust_link_handler",
             "deps": "none",  # disable gcc scandeps
@@ -273,7 +278,7 @@ def __step_config(ctx, step_config):
         {
             "name": "rust_rlib",
             "action": "(.*_)?rust_rlib",
-            "inputs": rust_toolchain,
+            "inputs": rust_inputs,
             "indirect_inputs": rust_compile_indirect_inputs,
             "deps": "none",  # disable gcc scandeps
             "remote": remote,
@@ -283,7 +288,7 @@ def __step_config(ctx, step_config):
         {
             "name": "rust_staticlib",
             "action": "(.*_)?rust_staticlib",
-            "inputs": rust_toolchain,
+            "inputs": rust_inputs,
             "indirect_inputs": rust_compile_indirect_inputs,
             "deps": "none",  # disable gcc scandeps
             "remote": remote,
@@ -314,7 +319,7 @@ def __step_config(ctx, step_config):
         {
             "name": "rust/clippy",
             "command_prefix": "python3 ../../build/rust/gni_impl/clippy_wrapper.py",
-            "inputs": rust_toolchain,
+            "inputs": rust_inputs,
             "indirect_inputs": rust_link_indirect_inputs,
             "remote": remote,
             "timeout": "2m",
@@ -325,7 +330,7 @@ def __step_config(ctx, step_config):
             # TODO(b/356496947): need to run scandeps?
             "name": "rust/bindgen",
             "command_prefix": "python3 ../../build/rust/gni_impl/run_bindgen.py",
-            "inputs": rust_toolchain + clang_inputs,
+            "inputs": rust_inputs + clang_inputs,
             "remote": False,
             "timeout": "2m",
         },
--- a/build/toolchain/gcc_toolchain.gni
+++ b/build/toolchain/gcc_toolchain.gni
@@ -712,7 +712,6 @@ template("single_gcc_toolchain") {
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $libname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_rlib") {
@@ -745,7 +744,6 @@ template("single_gcc_toolchain") {
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link,metadata=$rmetaname -o $rlibname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_bin") {
@@ -767,7 +765,6 @@ template("single_gcc_toolchain") {
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $exename -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS {{ldflags}} ${extra_ldflags} RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_cdylib") {
@@ -790,7 +787,6 @@ template("single_gcc_toolchain") {
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $dllname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS {{ldflags}} ${extra_ldflags} RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_dylib") {
@@ -815,7 +811,6 @@ template("single_gcc_toolchain") {
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link,metadata=$rmetaname -o $dllname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS {{ldflags}} ${extra_ldflags} RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
 
       tool("rust_macro") {
@@ -838,7 +833,6 @@ template("single_gcc_toolchain") {
             "$rustc_common_args {{rustdeps}} {{externs}} SOURCES {{sources}}"
         command = "$rustc_wrapper --depfile=$depfile --emit=dep-info=$depfile,link -o $dllname -- -Clinker=\"${invoker.cxx}\" {{source}} LDFLAGS {{ldflags}} ${extra_ldflags} RUSTENV {{rustenv}}"
         rust_sysroot = rust_sysroot_relative
-        inputs = rustc_wrapper_inputs
       }
     }
 
